### PR TITLE
infra: load action modal from ajax request

### PR DESCRIPTION
**Pull request by Mahaveer1141**
## What does this change do?
- load action modal from Ajax request

## Why is this change needed?
- too many modals load at once on the index.

## How were the changes done?
- added a new action to load modal.
- updated action dropdown in index for action details
- sending a ajax request on click of action button.

https://github.com/user-attachments/assets/c79bee59-030a-4a7d-963e-798b5ffefb4c

### CODE REVIEW

**Review by mikevic — APPROVED** [review]

File: app/assets/javascripts/cm_admin/custom_action.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    const actionName = $(this).attr("data-action_name");
+    const modelName = $(this).attr("data-model_name");
+    const recordId = $(this).attr("data-record_id");
+    const modalContainer = $(
+      "[data-behaviour='custom-action-modal-container']"
+    );
+    if (!actionName || !modelName || !recordId || !modalContainer) return;
+
+    $.ajax({
+      url: `/cm_admin/${modelName}/${recordId}/custom_action_modal/${actionName}`,
```
[the remark this answers is not in the document]
Comment (Mahaveer1141) [reply app/assets/javascripts/cm_admin/custom_action.js]: @mikevic I fixed this

File: app/controllers/cm_admin/resource_controller.rb [hunk 2]
```diff
@@ -172,6 +172,17 @@ def cm_custom_method(params)
       end
     end
 
+    def cm_custom_action_modal(params)
+      scoped_model = "CmAdmin::#{@model.name}Policy::Scope".constantize.new(Current.user, @model.name.constantize).resolve
+      @ar_object = fetch_ar_object(scoped_model, params[:id])
+      if params[:action_name] == 'destroy'
+        render partial: '/layouts/destroy_action_modal', locals: { ar_object: @ar_object }
+      else
+        custom_action = @model.available_actions.select { |x| x.name == params[:action_name].to_s }.first
```
Comment (anbublacky) [inline app/controllers/cm_admin/resource_controller.rb:181]: @Mahaveer1141 This variable is not used, Can we remove it

**Review by anbublacky — CHANGES_REQUESTED** [review]

File: app/controllers/cm_admin/resource_controller.rb (hunk 2 above)
Comment (Mahaveer1141) [reply app/controllers/cm_admin/resource_controller.rb:181]: @anbublacky this variable is being used in partial

Comment (anbublacky) [reply app/controllers/cm_admin/resource_controller.rb:181]: @Mahaveer1141 This is not a instance variable and not passed through locals, SO it is not used right?

Comment (Mahaveer1141) [reply app/controllers/cm_admin/resource_controller.rb:181]: @anbublacky this is being used, in this next line I am passing this to locals variable

**Review by anbublacky — APPROVED** [review]

**Pull request merged**